forecasttools/daily_to_epiweekly.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+
+import polars as pl
+
+
+# NOTE: this needs to change to ACTUAL epiweeks
+def calculate_epidate(date):
+    iso_calendar = date.isocalendar()
+    return iso_calendar[1], iso_calendar[0]
+
+
+def daily_to_epiweekly(
+    forecast_df: pl.DataFrame,
+    value_col: str = "value",
+    date_col: str = "date",
+    id_cols: list[str] = [".draw"],
+    weekly_value_name: str = "weekly_value",
+    strict: bool = False,
+) -> pl.DataFrame:
+    """
+    Aggregate daily forecast draws to epiweekly.
+    """
+    # check intended df columns are in received df
+    # forecast_df_cols = forecast_df.columns
+    # = [value_col, date_col] + id_cols
+    # assert set(required_cols).issubset(set(forecast_df_cols)),f"Column mismatch between require columns {required_cols} and forecast dateframe columns {forecast_df_cols}."
+    # add epiweek and epiyear columns
+    forecast_df = forecast_df.with_columns(
+        [
+            pl.col(date_col)
+            .map_elements(
+                lambda x: calculate_epidate(datetime.strptime(x, "%Y-%m-%d"))[
+                    0
+                ]
+            )
+            .alias("epiweek"),
+            pl.col(date_col)
+            .map_elements(
+                lambda x: calculate_epidate(datetime.strptime(x, "%Y-%m-%d"))[
+                    1
+                ]
+            )
+            .alias("epiyear"),
+        ]
+    )
+    # group by epiweek, epiyear, and the id_cols
+    group_cols = ["epiweek", "epiyear"] + id_cols
+    grouped_df = forecast_df.group_by(group_cols)
+    # number of elements per group
+    n_elements = grouped_df.agg(pl.count().alias("n_elements"))
+    # check if any week has more than 7 c dates
```

Comment:
```suggestion
    # check if any week has more than 7 dates
```
Typo?

Reply:
Yes, good catch + don't know why this wasn't flagged by `typos` hook in `pre-commit`, which has caught stray characters.